azurerm/resource_arm_virtual_machine_scale_set_test.go

```diff
@@ -24,9 +25,12 @@ func TestAccAzureRMVirtualMachineScaleSet_basic(t *testing.T) {
 				Config: config,
 				Check: resource.ComposeTestCheckFunc(
 					testCheckAzureRMVirtualMachineScaleSetExists("azurerm_virtual_machine_scale_set.test"),
-
-					// single placement group should default to true
+					resource.TestCheckResourceAttrSet("azurerm_virtual_machine_scale_set.test", "vm_hostnames"),
+					resource.TestCheckResourceAttrSet("azurerm_virtual_machine_scale_set.test", "vm_statuses"),
+					resource.TestCheckResourceAttrSet("azurerm_virtual_machine_scale_set.test", "vm_primary_private_ip_addresses"),
+					// testing default scaleset values
 					testCheckAzureRMVirtualMachineScaleSetSinglePlacementGroup("azurerm_virtual_machine_scale_set.test", true),
+					testCheckAzureRMVirtualMachineScaleSetIPForwarding("azurerm_virtual_machine_scale_set.test", false),
```

Comment:
For my personal knowledge, what is the difference between the current test VS checking for the state only?  The current methods queries the API?

Reply:
After review, indeed the diff is an API call on the remote resource VS what terraform stored in its state. 

I don't mind converting to `resource.TestCheckResourceAttr` this will probably speed up the test

Reply:
Thanks! 

The reason for this is we want to verify that the create/update and read functions. If we call out to the API we are only checking one direction, and not ensuring that the provider read it back (or imported) into the state correctly.